### PR TITLE
Frontstage changes to support multiple bussiness and surveys share

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?
Frontstage changes to support multiple bussiness and surveys share
# How to test?

# Trello
https://trello.com/c/bpABDVK3/595-s33-user-account-creation-share-surveys-enable-multiple-business-sharing-story-6

### CODE REVIEW

File: frontstage/templates/surveys/surveys-share/send-instructions.html [hunk 1]
```diff
@@ -50,16 +50,18 @@
 {% block main %}
 <h1 class="u-fs-xl">Send instructions</h1>
 <p>We will send an email to <b>{{ email }}</b> with instructions to access the following surveys:</p>
-<h2 class="u-fs-m">{{ business_name }}</h2>
-{% if surveys|length > 0 %}
+{% for key in share_dict %}
```
Comment (insacuri) [inline frontstage/templates/surveys/surveys-share/send-instructions.html:53]: What is the `key` representing here?  A business?  Would be nice to replace this name for something more specific

File: frontstage/views/account/account_survey_share.py [hunk 2]
```diff
@@ -23,8 +23,7 @@
 @account_bp.route('/share-surveys', methods=['GET'])
 @jwt_authorization(request)
 def share_survey_overview(session):
-    flask_session.pop('share_survey_business_selected', None)
-    flask_session.pop('share_survey_surveys_selected', None)
+    flask_session.pop('share', None)
```
Comment (insacuri) [inline frontstage/views/account/account_survey_share.py:26]: Can this have a better name? A key of 'share' in the session doesn't say a huge amount about what it's for

File: frontstage/views/account/account_survey_share.py [hunk 3]
```diff
@@ -44,38 +42,42 @@ def share_survey_business_select(session):
...
+    failed_survey_key = request.args.get('key', None)
     return render_template('surveys/surveys-share/survey-select.html',
-                           surveys=surveys, business_name=selected_business[0]['name'],
-                           is_max_share_survey=is_max_share_survey)
+                           share_dict=share_dict, is_max_share_survey=is_max_share_survey,
+                           failed_survey_key=failed_survey_key)
 
 
-def validate_max_shared_survey():
+def validate_max_shared_survey(key: str, share_survey_surveys_selected: list):
```
Comment (insacuri) [inline frontstage/views/account/account_survey_share.py:70]: can `key` just be `business_id` if that's what it is?  It makes it easier to read

File: frontstage/views/account/account_survey_share.py [hunk 4]
```diff
@@ -149,17 +158,16 @@ def build_payload(respondent_id):
...
     pending_shares = []
-    for survey in share_survey_surveys_selected:
-        pending_share = {'business_id': business_id,
-                         'survey_id': survey,
-                         'email_address': email,
-                         'shared_by': respondent_id}
-        pending_shares.append(pending_share)
-
+    share_dictionary = flask_session['share']
+    for key in share_dictionary:
```
Comment (insacuri) [inline frontstage/views/account/account_survey_share.py:164]: Can key be changed to what it is?  I think it's a business_id so it should be called that

File: frontstage/templates/surveys/surveys-share/send-instructions.html [hunk 5]
```diff
@@ -50,16 +50,18 @@
...
+    <h2 class="u-fs-m">{{ key }}</h2>
+    {% if share_dict[key]|length > 0 %}
```
Comment (insacuri) [inline frontstage/templates/surveys/surveys-share/send-instructions.html:55]: Need spaces around the pipe

File: frontstage/templates/surveys/surveys-share/survey-select.html [hunk 6]
```diff
@@ -70,22 +70,25 @@
 {% endif %}
 
 <h1>Which surveys do you want to share?</h1>
-{% set checkboxData = [] %}
 <form action="" method="post">
     <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
+    {% for key in share_dict %}
```
Comment (insacuri) [inline frontstage/templates/surveys/surveys-share/survey-select.html:75]: Can key be changed to better describe what it is?

**Review by insacuri — CHANGES_REQUESTED** [review]
Minor changes but the code looks good

File: frontstage/templates/surveys/surveys-share/send-instructions.html (hunk 1 above)
Comment (creativeappdevamit) [reply frontstage/templates/surveys/surveys-share/send-instructions.html:53]: 👍

File: frontstage/views/account/account_survey_share.py (hunk 2 above)
Comment (creativeappdevamit) [reply frontstage/views/account/account_survey_share.py:26]: 👍

File: frontstage/views/account/account_survey_share.py (hunk 3 above)
Comment (creativeappdevamit) [reply frontstage/views/account/account_survey_share.py:70]: 👍 +1

File: frontstage/views/account/account_survey_share.py (hunk 4 above)
Comment (creativeappdevamit) [reply frontstage/views/account/account_survey_share.py:164]: 👍

File: frontstage/templates/surveys/surveys-share/send-instructions.html (hunk 5 above)
Comment (creativeappdevamit) [reply frontstage/templates/surveys/surveys-share/send-instructions.html:55]: 👍

File: frontstage/templates/surveys/surveys-share/survey-select.html (hunk 6 above)
Comment (creativeappdevamit) [reply frontstage/templates/surveys/surveys-share/survey-select.html:75]: 👍

**Review by insacuri — APPROVED** [review]

**Pull request merged**